### PR TITLE
fix(feature/cvsb-18389): trigger reference data api calls when the app start button is clicked

### PR DESCRIPTION
**Pull request by alleypa**
trigger reference data api calls when the app start button is clicked

## Ticket title
One line description
[link to ticket number](https://jira.dvsacloud.uk/browse/CVSB-18389)

## Checklist
- [x] Code has been tested manually
- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/pages/test-station/test-station-home/test-station-home.ts [hunk 1]
```diff
@@ -79,14 +77,21 @@ export class TestStationHomePage implements OnInit {
     this.firebaseLogsService.setScreenName(FIREBASE_SCREEN_NAMES.GET_STARTED);
   }
 
-  getStarted(): void {
+  async getStarted() {
+    let err: Error, IsDataSynced: boolean;
+    [err, IsDataSynced] = await this.syncService.startSync();
```
Comment (alex-deloitte) [inline src/pages/test-station/test-station-home/test-station-home.ts:82]: Why do you need the error parameter here? I can't see where it is used

Comment (alleypa) [reply src/pages/test-station/test-station-home/test-station-home.ts:82]: It is the first (potential) response parameter from calling this.syncService.startSync(). Not necessary for it to be used within the method. But more for documentation purpose.

File: src/pages/test-station/test-station-home/test-station-home.ts [hunk 2]
```diff
@@ -79,14 +77,21 @@ export class TestStationHomePage implements OnInit {
...
+
+    if (IsDataSynced) {
+      this.setPage();
+    }
```
Comment (bristol-mike-jones) [inline src/pages/test-station/test-station-home/test-station-home.ts:86]: Need to have an else here as an insurance policy log statement that sticks a logs out via the authenticated logs.  Saying something about and unexpected error trying to sync the data

If the err object is anything other than undefined the err object should be included in the log message.

**Review by alex-deloitte — APPROVED** [review]

**Comment by JonLazariniDD** [comment]
Looks good, since we know that the Ionic storage might take some time process a large amount of data received from the services, we created a subsequent ticket https://jira.dvsacloud.uk/browse/CVSB-18277 where we will have to also tackle how we manage errors since reference data is a must to proceed.

**Review by JonLazariniDD — APPROVED** [review]

**Pull request merged**